### PR TITLE
Fix DistrictHeating:Water in HVACTemplate:Plant:Boiler

### LINKED ISSUE #10819: ExpandObjects is still writing DistrictHeating instead of DistrictHeating:Water
Issue overview
--------------
For HVACTemplate:Plant:Boiler with Boiler Type = DistrictHotWater, ExpandObjects is writing "DistrictHeating" instead of "DistrictHeating:Water".

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus 24.2
 - Helpdesk ticket number 16795

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added EnergyPlusDevSupport\DefectFiles
 - [ ] Ticket added to EnergyPlus Defect Complexity (Github Project)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
 - Fixes #10819 
 
### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by rraustad** [comment]
Not to add to the issue but I did a search of the repository. Not sure why epfilter.f90 was not included in that search?

image

**Comment by mjwitte** [comment]
@rraustad Good catch. Found another one in the same doc.

File: doc/getting-started/src/tutorial-example-for-running-energyplus/instructions.tex [hunk 1]
```diff
@@ -28,7 +28,7 @@ \subsection{Exercise 1A. Run Pre-Defined Building with no Windows}\label{exercis
 \item
   Outdoor Dry Bulb -- is being reported (so you can compare to outside temperature)
 \item
-  The meter for the heating in the facility - DistrictHeating:Facility -- is being reported. Facility is the entire building.
+  The meter for the heating in the facility - DistrictHeatingWater:Facility -- is being reported. Facility is the entire building.
```
Comment (rraustad) [inline doc/getting-started/src/tutorial-example-for-running-energyplus/instructions.tex:31]: I didn't know this was wrong. What about this?

image

Comment (rraustad) [reply doc/getting-started/src/tutorial-example-for-running-energyplus/instructions.tex:31]: Oh never mind, that's a comment.

**Comment by rraustad** [comment]
Is this an issue?

image

**Comment by mjwitte** [comment]
@rraustad Thanks for more sleuthing. Added some more cleanups (and potentially real fixes).

File: src/EnergyPlus/Plant/Component.hh [hunk 2]
```diff
@@ -158,7 +158,8 @@ namespace DataPlant {
         false, //	"AirTerminal:SingleDuct:ConstantVolume:FourPipeBeam"
         false, //	"ZoneHVAC:CoolingPanel:RadiantConvective:Water"
         false, //	"HeatPump:PlantLoop:EIR:Cooling"
-        false  //	"HeatPump:PlantLoop:EIR:Heating"
+        false, //	"HeatPump:PlantLoop:EIR:Heating"
+        false  //   "DistrictHeating:Steam"
```
Comment (mjwitte) [inline src/EnergyPlus/Plant/Component.hh:162]: This array should be parallel with [`PlantEquipmentType`](https://github.com/NREL/EnergyPlus/blob/68508860518b5f36a81d9ddeb28cc661f229b631/src/EnergyPlus/Plant/Enums.hh#L108) enum. It was missing a slot for `PurchSteam` at the end.
https://github.com/NREL/EnergyPlus/blob/68508860518b5f36a81d9ddeb28cc661f229b631/src/EnergyPlus/Plant/Enums.hh#L209-L211

File: src/EnergyPlus/PollutionModule.hh [hunk 3]
```diff
@@ -196,8 +196,8 @@ namespace Pollution {
         PollFuel::OtherFuel1,  // OtherFuel1
         PollFuel::OtherFuel2,  // OtherFuel2
         PollFuel::Electricity, // DistrictCooling
-        PollFuel::NaturalGas,  // DistrictHeating
-        PollFuel::NaturalGas,  // Steam
+        PollFuel::NaturalGas,  // DistrictHeatingWater
+        PollFuel::NaturalGas,  // DistrictHeatingSteam
```
Comment (mjwitte) [inline src/EnergyPlus/PollutionModule.hh:200]: Ignore the commit comment about Pollution. This was must a comment cleanup.

**Review by rraustad — APPROVED** [review]
@Myoldmopar this looks to be a good clean up of a previous change.

**Review by Myoldmopar — APPROVED** [review]
Looks good to me.

**Comment by Myoldmopar** [comment]
And it's happy locally.  Merging.  Thanks @mjwitte

**Pull request merged**